code/controllers/subsystem/economy.dm

```diff
@@ -0,0 +1,309 @@
+
+	if(!paid_for && !(order in requestlist))
+		requestlist += order //submit a request but do not finalize it
+		return TRUE
+
+	if(order.requires_head_approval || order.requires_qm_approval)
+		return TRUE
+
+	//if purchaser has already paid it means it's fully approved, finalize order
+	if(paid_for)
+		finalize_supply_order(order) //if payment was succesful, add order to shoppinglist
+		return TRUE
+	//we shouldn't be here, this means that the crate isn't paid for and doesn't need approval
+	qdel(order) //only the strong will survive
+	return FALSE
+
+/datum/controller/subsystem/economy/proc/finalize_supply_order(datum/supply_order/order)
+	if(!order)
+		CRASH("finalize_supply_order() called with a null datum/supply_order")
+	if(order in requestlist)
+		requestlist -= order
+
+	if(SSshuttle.supply.getDockedId() == "supply_away" && SSshuttle.supply.mode == SHUTTLE_IDLE)
+		deliverylist += order
+	else
+		shoppinglist += order
+
+////////////////////////////
+/// Paycheck Stuff /////////
+////////////////////////
+
+/datum/controller/subsystem/economy/proc/payday()
+	payday_count++
+	var/total_payout = 0
+	var/total_accounts = 0
+	var/datum/money_account_database/main_station/station_db = GLOB.station_money_database
+	var/list/all_station_accounts = station_db.user_accounts + station_db.get_all_department_accounts()
+	for(var/datum/money_account/account in all_station_accounts)
+		var/amount_to_pay = account.payday_amount + global_paycheck_bonus - global_paycheck_deducation
+		if(length(account.pay_check_bonuses))
+			for(var/bonus in account.pay_check_bonuses)
+				amount_to_pay += bonus
+				LAZYREMOVE(account.pay_check_bonuses, bonus)
+		for(var/deduction in account.pay_check_deductions)
+			amount_to_pay = max(amount_to_pay - deduction, 0)
+			LAZYREMOVE(account.pay_check_deductions, deduction)
+		station_db.credit_account(account, amount_to_pay, "Payday", "NAS Trurl Payroll", FALSE)
+		if(account.account_type == ACCOUNT_TYPE_PERSONAL)
+			if(LAZYLEN(account.associated_nanobank_programs))
+				for(var/datum/data/pda/app/nanobank/program as anything in account.associated_nanobank_programs)
+					program.announce_payday(amount_to_pay)
+		total_accounts++
+		total_payout += amount_to_pay
+
+	//reset global paycheck modifiers to 0
+	global_paycheck_bonus = 0
+	global_paycheck_deducation = 0
+	//update space credit statistics
+	space_credits_created += total_payout
+	total_space_credits += total_payout
+	//alert admins and c*ders alike
+	log_debug("Payday Count: [payday_count] - [total_payout] credits paid out to [total_accounts] accounts")
+	message_admins("Here comes the money! Payday: [total_payout] credits paid out to [total_accounts] accounts")
+
+
+//Called by the gameticker
+/datum/controller/subsystem/economy/proc/process_job_tasks()
+	for(var/mob/M in GLOB.player_list) //why not just make a global list of players with job objectives???? someone else fix this ~sirryan
+		if(M.mind)
```

Comment:
Can you turn this into an early continue?

Reply:
done